bin/build_utils

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+set -euo pipefail
+
+####
+# Functions to generate version numbers for this project
+####
+
+readonly VERSION_GO_FILE="pkg/trace/version.go"
```

Comment:
Embedding the version number in a go file especially several directories deep in the module means tracking and updating the version number in 2 places and makes autoreleases in the near future more difficult.  The Jenkinsfile currently creates a VERSION file in the project root that has the calculated version (`changelog-build_number`), su unless there's a pressing need for the version number of the library available at runtime, I'd suggest the following:

And then update uses of `VERSION_GO_FILE` to just be `VERSION_FILE`
```suggestion
# Set up VERSION file for local development
if [ ! -f "VERSION" ]; then
  echo -n "0.0.dev" > VERSION
fi
readonly VERSION_FILE="$(<VERSION)"
```